Classes/Controller/SearchController.php

```diff
@@ -113,6 +123,11 @@ public function resultsAction()
             ];
 
             $values = $this->emitActionSignal(__CLASS__, __FUNCTION__, [$values]);
```

Comment:
Is this signal unusable for the things you want?
If there is not all what you want, pack this in array.
Would propose to extend all the signals and pack the stuff in the docs and examples.

Reply:
Main difference is that Data Processing is configured via typoscript, allowing to plug different configs in places.
Instead of Signals which are global.

Reply:
No, i mean that we could plug this stuff via this signal[s], via something like "Facade/Gateway" similar design pattern.

Reply:
The `ContentDataProcessor` only needs the `ContentObjectRenderer` which we can get pretty much anywhere injection works. And the configuration from Typoscript of course.
So yeah we could plug it via the signals.

Reply:
FYI: signal/slot is deprecated
https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/Hooks/EventDispatcher/Index.html#eventdispatcher